File: USACO/Competitions/2016-2017/US Open Silver/cownomics.java
import java.io.*;
import java.util.Scanner;
public class cownomics {
    public static void main(String[] args) throws IOException {
        Scanner input = new Scanner(new File("cownomics.in"));
        int N = input.nextInt(), M = input.nextInt();
        String[] spotty = new String[N], plain = new String[N];
        for (int i = 0; i < N; i++) spotty[i] = input.next();
        for (int i = 0; i < N; i++) plain[i] = input.next();
        input.close();
        int count = 0;
        for (int i = 0; i < M-2; i++) {
            for (int j = i+1; j < M-1; j++) {
                outerloop: for (int k = j+1; k < M; k++) {
                    int[] marked = new int[64];
                    for (String s: spotty) {
                        marked[conv(s.charAt(i))*16+conv(s.charAt(j))*4+conv(s.charAt(k))] = 1;
                    }
                    for (String s: plain) {
                        if (marked[conv(s.charAt(i))*16+conv(s.charAt(j))*4+conv(s.charAt(k))] == 1) continue outerloop;
                    }
                    count += 1;
                }
            }
        }
        // System.out.println(count);
        PrintWriter output = new PrintWriter(new File("cownomics.out"));
        output.println(count);
        output.close();
    }

    public static int conv(char c) {
        if (c == 'A') return 0;
        if (c == 'C') return 1;
        if (c == 'G') return 2;
        return 3;
    }
}

File: USACO/Competitions/2016-2017/US Open Silver/cownomics.py
with open('cownomics.in') as f:
    N, M = map(int, f.readline().split(" "))
    spotty = [f.readline().strip("\n") for _ in range(N)]
    plain = [f.readline().strip("\n") for _ in range(N)]

# print(N, M)
# print(spotty)
# print(plain)

def conv(n):
    return {'A': 0, 'C': 1, 'G': 2, 'T': 3}[n]

count = 0
for i in range(M-2):
    for j in range(i+1, M-1):
        for k in range(j+1, M):
            marked = [0] * 64
            for s in spotty:
                marked[16*conv(s[i])+4*conv(s[j])+conv(s[k])] = 1
            shouldBreak = False
            for p in plain:
                if marked[16*conv(p[i])+4*conv(p[j])+conv(p[k])]:
                    shouldBreak = True
                    break
            if shouldBreak: continue
            count += 1

with open('cownomics.out', 'w') as f:
    # print(count)
    f.write("%s\n" % count)